url-parameters/tests.json

```diff
@@ -1,55 +1,124 @@
 {
     "trackingParameters": {
-        "name": "Tracking parameters are able to be removed from a URL",
+        "name": "Tracking parameters protection",
         "desc": "Those tests use config_reference.json",
         "referenceConfig": "config_reference.json",
         "tests": [
             {
-                "name": "Remove utm tracking parameters from URL",
-                "testURL": "http://www.example.com/test.html?utm_source=test&utm_medium=test&utm_campaign=test&utm_term=test&utm_content=test",
-                "expectURL": "http://www.example.com/test.html",
+                "name": "Non-tracking parameters are not removed",
+                "testURL": "http://www.example.com/test.html?q=test&id=5",
+                "expectURL": "http://www.example.com/test.html?q=test&id=5",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Parameters with values that match tracking parameter names are not removed",
+                "testURL": "http://www.example.com/test.html?a=utm_test&b=regexp&c=gclid&d=fbclid",
+                "expectURL": "http://www.example.com/test.html?a=utm_test&b=regexp&c=gclid&d=fbclid",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Plain-text tracking paramters are removed",
+                "testURL": "http://example.com/?a=123&gclid=test&fbclid=value&b=456",
+                "expectURL": "http://example.com/?a=123&b=456",
                 "exceptPlatforms": []
             },
             {
-                "name": "Remove fbclid parameter and retain other parameters",
+                "name": "When plain-text tracking parameters are removed, other parameters are preserved",
                 "testURL": "http://www.example.com/test.html?fbclid=test&q=test&id=5",
                 "expectURL": "http://www.example.com/test.html?q=test&id=5",
                 "exceptPlatforms": []
             },
             {
-                "name": "Retain all non tracking parameters",
-                "testURL": "http://www.example.com/test.html?q=test&id=5",
-                "expectURL": "http://www.example.com/test.html?q=test&id=5",
+                "name": "Parameters similar to plain-text tracking parameters are not removed",
+                "testURL": "http://www.example.com/test.html?fbclid1=test",
+                "expectURL": "http://www.example.com/test.html?fbclid1=test",
                 "exceptPlatforms": []
             },
             {
-                "name": "Don't remove parameters because domain is in exceptions list",
-                "testURL": "http://example2.com/test.html?gclid=test&fbclid=test",
-                "expectURL": "http://example2.com/test.html?gclid=test&fbclid=test",
+                "name": "Parameters case insensitively matching plain-text tracking parameters are not removed",
```

Comment:
oh, that's interesting, I'd expect them to be removed I think. @SlayterDev what's your take? Is that specified in the spec?

Reply:
I figured case-sensitive matching was implied by the `"utm_[a-zA-Z]*"` trackingParameters entry in extension-config.json. FWIW I also think that is the right call, since case-sensitive matching could turn out to be useful in the future.

Reply:
Also, if we do plain-text parameter matching case-sensitively that matches how the declarativeNetRequest API's QueryTransform conditions work. That will make porting the feature to mv3 easier.

Reply:
We can imply case sensitivity. FWIW I've never seen a parameter that isn't just lowercase letters.